tests/ignite/metrics/test_loss.py

```diff
@@ -7,8 +8,83 @@
 from torch.nn.functional import nll_loss
 
 import ignite.distributed as idist
+from ignite.engine import State
 from ignite.exceptions import NotComputableError
-from ignite.metrics import Loss
+from ignite.metrics import Loss, Precision
+
+
+class DummyLoss1(Loss):
+    def __init__(self, loss_fn, true_output, output_transform=lambda x: x):
+        super(DummyLoss1, self).__init__(loss_fn, output_transform=output_transform)
+        print(true_output)
+        self.true_output = true_output
+
+    def reset(self):
+        pass
+
+    def compute(self):
+        pass
+
+    def update(self, output):
+
+        assert output == self.true_output
+
+
+def test_output_as_mapping_wrong_keys():
+    loss_metric = DummyLoss1(nll_loss, true_output=(0, 1, {}))
+    state = State(output=({"y1": 0, "y2": 1, "kwargs": {}}))
+    engine = MagicMock(state=state)
+
+    with pytest.raises(
+        ValueError,
+        match=r"When transformed engine's output is a mapping, "
+        r"it should contain \('y_pred', 'y', 'criterion_kwargs'\) keys",
+    ):
+        loss_metric.iteration_completed(engine)
+
+
+def test_output_as_mapping_keys_is_none():
+    class DummyLoss(Loss):
+        required_output_keys = None
+
+        def reset(self):
+            pass
+
+        def compute(self):
+            pass
+
+        def update(self, output):
+            pass
+
+    loss_metric = DummyLoss(nll_loss)
+    assert loss_metric.required_output_keys is None
+    state = State(output=({"y1": 0, "y2": 1, "kwargs": {}}))
+    engine = MagicMock(state=state)
+
+    with pytest.raises(TypeError, match=r"Transformed engine output for DummyLoss metric should be a tuple/list"):
+        loss_metric.iteration_completed(engine)
```

Comment:
Thanks for updating the tests. I think we do not need those tests:
- test_output_as_mapping_wrong_keys
- test_output_as_mapping_keys_is_none
As it is already covered in test_metric.py...

Others look ok.

Reply:
Ok will make the changes